birdhouse/scripts/read-configs.include.sh

```diff
@@ -166,19 +177,40 @@ read_components_default_env() {
 
     # EXTRA_CONF_DIRS and DEFAULT_CONF_DIRS relative paths are relative to COMPOSE_DIR.
     if [ -d "$COMPOSE_DIR" ]; then
-        cd "$COMPOSE_DIR"
+        cd "$COMPOSE_DIR" >/dev/null
     fi
 
     source_conf_files "$DEFAULT_CONF_DIRS" 'DEFAULT_CONF_DIRS'
     source_conf_files "$EXTRA_CONF_DIRS" 'EXTRA_CONF_DIRS'
 
     # Return to previous pwd.
     if [ -d "$COMPOSE_DIR" ]; then
-        cd -
+        cd - >/dev/null
     fi
 }
 
 
+check_optional_vars() {
+
+    for i in ${OPTIONAL_VARS}
+    do
+        v="${i}"
+        d=`eval echo "$v"`
+        n="${i#\$}"
+        default="\${__DEFAULT__${n}}"
+        result=`echo "${d}" | grep -c "${default}"`
+        if [ -z "`eval "echo ${v}"`" ]
+        then
+            echo "${MSG_WARN}Optional variable [${n}] is not set. Check env.local file."
+        fi
+        if [ "${result}" -gt 0 ]
+        then
+            echo "${MSG_WARN}Optional variable [${n}] employs a default recommended for override. Check env.local file."
```

Comment:
Interesting !   We probably need to add all default passwords to the list of "recommended for override"

Reply:
Yes. Very good point!

Reply:
After looking at the code a bit more, I'm not certain what to do regarding "recommended overrides" for passwords. 
The only place where their values are specified is in the `env.local.example` for illustration purpose.
If the user runs `pavics-compose` without explicitly setting them in their `env.local`, the required `VARS` would error out with those missing definitions. I think it is safer to avoid setting defaults to make the failing error obvious and breaking rather than possibility missed through a warning.
 
The only issue I can foresee is with the vagrant script that does a `cp env.local.example env.local`, which could end up using the defaults. But in that case, is it really an issue, given that this copy results from a lack of overrides anyway?

Reply:
I think that a user might do `cp env.local.example env.local` as a starting point when setting up their node and then forget to change the passwords from the defaults (or get lazy and just keep the examples because it's easier). 

We could use a similar mechanism to check if the password is the same as the example in `env.local.example` and show a warning/error if they don't change them. For security purposes, all passwords should really be changed anyway.

Reply:
Ok what if we do something like this. For each of the "password" and "secret" variables that really shouldn't be the defaults we add (in default.env):

```
__DEFAULT__MAGPIE_ADMIN_PASSWORD='qwertyqwerty!'
__DEFAULT__GRAFANA_ADMIN_PASSWORD='changeme!'
# etc.
```

but we don't set these defaults in default.env like we do with the optional variables.

then in read-configs.include.sh we have another loop like:

```
for i in ${VARS}
do
  # if the current value matches the default 
  log WARN "Variable [${n}] employs a default recommended for override. The security of your deployment may be compromised unless it is changed. Check env.local file."
done
```

This would only apply to variables for which we've created a `__DEFAULT__*` equivalent. If the variable didn't have a value, a previous error message would have already told the user that.

Reply:
Yes, we could do that, since the real variables would remain without explicit defaults.

Reply:
This is now added.